src/lib.rs

```diff
@@ -366,4 +375,26 @@ mod contract {
             }
         }
     }
+
+    impl ToStr for crate::types::NonceError {
+        fn to_str(&self) -> &str {
+            match self {
+                Self::NonceOverflow => "ERR_NONCE_OVERFLOW",
+                Self::IncorrectNonce => "ERR_INCORRECT_NONCE",
```

Comment:
Speaking of this, what is our typical Error string? In Rust, you typically would have it as all lower case as when it panics on screen, the error reads out like a grammatical sentence where as if it is a panic, it would be a written sentence. Of course with the SDK it probably is something different and spits out to a log somewhere, but I never understood why we would detract from that as it becomes a new thing for Rust developers to pick up on, which we do differently than every other Rust lib.

Reply:
@joshuajbouw These are not error strings, they are error codes. It isn't humans interacting with the contract, it is machines. Libraries (such as Aurora.js) and tools (such as the Aurora CLI) will need to translate the codes into error strings; the contract's error handling should merely focus on conciseness and unambiguity in parsing flagged error conditions.